GCRCatalogs/dc2_matched_table.py

```diff
@@ -5,66 +5,147 @@
 based on a matching table provided by Javier Sanchez .
```

Comment:
Is the class meant to deal with any matched catalog, or only the dpdd to truth matching that seems to be targeted here? It may be useful to expand the docstring to provide more information to the user as to how to use this class.

Reply:
@johannct  This class is meant to deal with the matched tables for DC2 object catalogs provided by Javi. I am not aware of any other possible application at present. I have expanded the doc string to provide more description.